common/web/keyboard-processor/src/text/kbdInterface.ts

```diff
@@ -529,6 +529,43 @@ namespace com.keyman.text {
       }
     }
 
+    /**
+     * Maps a KeyEvent's modifiers to their appropriate value for key-rule evaluation
+     * based on the rule's specified target modifier set.
+     *
+     * Mostly used to correct chiral OSK-keys targeting non-chiral rules.
+     * @param e The source KeyEvent
+     * @returns
+     */
+    private static getEventRuleModifiers(e: KeyEvent, targetModifierMask: number): number {
```

Comment:
Given we only ever use `e.Lmodifiers` in the `KeyEvent`, why don't we make this function more "generic" in nature and pass in that value? Rename it to `transformModifierChiralityModeForTarget` or something like that?

```suggestion
    private static getEventRuleModifiers(eventModifiers: number, targetModifierMask: number): number {
```